File: abc/159/b.py
# definitioin function palindromeChecker()
def palindromeChecker(S, N): # N is required to be odd.
  flag = True
  half = N // 2
  for i in range(half):
    if S[i] != S[-(i+1)]:
      flag = False
  return flag

def main():
  # input
  S = str(input())

  # compute
  N = len(S)
  T = S[:(N-1)//2]
  M = len(T)
  U = S[(N+3)//2 - 1:]
  O = len(U)
  a = palindromeChecker(S, N)
  b = palindromeChecker(T, M)
  c = palindromeChecker(U, O)

  # output
  if a and b and c:
    print('Yes')
  else:
    print('No')


if __name__ == "__main__":
    main()

File: abc/159/b.c
// Now Underconstructing ...

#include <stdio.h>
#include <string.h>

int palindromeChecker(char* S, int N);

int main(void) {
  // input
  char S[100]; scanf("%s", &S[0]);

  // compute
  int N = strlen(S);
  int a = palindromeChecker(&S[0], N);

  // output
  if (a && b && c) {
    printf("Yes\n");
  }
  else {
    printf("No\n");
  }
  return 0;
}


// define function palindromeChecker()
int palindromeChecker(char* S, int N) { // N is required to be odd.
  int half = N/2, flag = 1, i;

  // compute
  for (i = 0; i < half; ++i) {
    if (S[i] != S[N-1-i]) {
      flag = 0;
    }
  }

  // output
  return flag;
}
